packages/keyring/src/pair/index.ts

```diff
@@ -113,23 +113,28 @@ export function createPair ({ toSS58, type }: Setup, { publicKey, secretKey }: P
   };
 
   const encodeAddress = (): string => {
-    const raw = TYPE_ADDRESS[type](publicKey);
+    // Need to be able to add from an eth address
+    if (type === 'ethereum' && publicKey.length === 20) {
+      return u8aToHex(publicKey);
+    } else {
+      const raw = TYPE_ADDRESS[type](publicKey);
```

Comment:
This is the logic that should be in the `TYPE_ADDRESS[type](...)` function - if that receives 20 bytes as input into that function, then treat it as such.

Reply:
TYPE_ADDRESS[type](...) dosn't leave room for flexibility. Please make a PR with your suggested changes

Reply:
Possibly after parachains have launched on Kusama, don't have much playing time atm.

Reply:
This is not playing it's important work. This change is very important to Moonbeam. We have been trying to get MetaMask compatibility for a long time now. Please include this change that is tested and we can refactor it together when you have more time

Reply:
Ok. So... I'm not trying to get you upset here.

You asked for a PR from me, I mentioned where my time is spent. I sadly cannot drop what I'm busy with atm. 

I understand it is important to your team, not diminishing the effort at all, it has been great and moved things along greatly. However, anything that stands in the way to parachains on Polkadot so we can pull all of this effort together, is "playtime" which don't have as much as I wish.

So... for best-effort, would suggest we cleanup this PR here, remove the duplication and use the functions as intended - even if they may need to change to cater for this. If you want me to attack this rather than changing this PR, as you asked for here, it will need to wait until there is a gap.

Reply:
I mean it is a change of `ethereum: (p: Uint8Array) => keccakAsU8a(secp256k1Expand(p)),` to `ethereum: (p: Uint8Array) => p.length === 20 ? p : keccakAsU8a(secp256k1Expand(p)),`

1. function used as intended
2. no duplication where used

Reply:
Ok sorry I thought it would be easier this way. I pushed a commit to use the function as intended